cpu/kinetis/periph/mcg.c

```diff
@@ -176,6 +176,66 @@ static void kinetis_mcg_init_mcgirclk(void)
     }
 }
 
+/**
+ * @brief   Internal helper for performing the actual auto-trim of the reference oscillators
+ *
+ * This is called internally by @ref kinetis_mcg_auto_trim_irc
+ */
+static int kinetis_mcg_auto_trim_exec(unsigned fast_irc)
+{
+    uint32_t expected_count = (CLOCK_BUSCLOCK * 21u);
+    if (fast_irc) {
+        bit_set8(&MCG->SC, MCG_SC_ATMS_SHIFT);
+        /* Fast IRC is expected to run at 4 MHz */
+        /* The 64 bit multiplication will be optimized away and replaced by a constant */
+        expected_count = ((uint64_t)expected_count * 128) / 4000000ul;
+    }
+    else {
+        bit_clear8(&MCG->SC, MCG_SC_ATMS_SHIFT);
+        expected_count /= 32768;
+    }
+    if (expected_count > 0xffff) {
+        /* Something is very wrong with the configuration */
+        return -2;
+    }
+    /* Load expected counts */
+    MCG->ATCVL = expected_count & 0xff;
+    MCG->ATCVH = (expected_count >> 8) & 0xff;
+    /* initiate auto trim procedure */
+    bit_set8(&MCG->SC, MCG_SC_ATME_SHIFT);
+    /* ATME bit will de-assert automatically when the procedure completes */
+    while (MCG->SC & MCG_SC_ATME_MASK) {}
```

Comment:
Is there a need for a timeout or will this always end or is it acceptable to potentially wait forever?